hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/MarkerBasedRollbackStrategy.java

```diff
@@ -91,4 +101,83 @@ public List<HoodieRollbackRequest> getRollbackRequests(HoodieInstant instantToRo
       throw new HoodieRollbackException("Error rolling back using marker files written for " + instantToRollback, e);
     }
   }
+
+  protected HoodieRollbackRequest createRollbackRequestForCreateMerge(String fileId,
+                                                                      String partitionPath,
+                                                                      StoragePath filePath,
+                                                                      HoodieInstant instantToRollback) {
+    if (table.version().greaterThanOrEquals(HoodieTableVersion.EIGHT)) {
+      return new HoodieRollbackRequest(partitionPath, fileId, instantToRollback.getTimestamp(),
+          Collections.singletonList(filePath.toString()), Collections.emptyMap());
+    } else {
+      String baseInstantTime = null;
+      if (FSUtils.isBaseFile(filePath)) {
+        HoodieBaseFile baseFileToDelete = new HoodieBaseFile(filePath.toString());
+        fileId = baseFileToDelete.getFileId();
+        baseInstantTime = baseFileToDelete.getCommitTime();
```

Comment:
I'm assuming this logic is copied from legacy version.

Reply:
yes.